src/arrayMethodSort.js

```diff
@@ -3,9 +3,34 @@
 /**
  * Implement method Sort
  */
+const defaultCompare = function (a, b) {
+  const aStr = String(a);
+  const bStr = String(b);
+
+  if (aStr < bStr) {
+    return -1;
+  }
+
+  if (aStr > bStr) {
+    return 1;
+  }
+
+  if (aStr === bStr) {
+    return 0;
+  }
+};
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction = defaultCompare) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
```

Comment:
You can optimize the inner loop to run only until 'this.length - i - 1' instead of 'this.length - 1'. This small change will reduce the number of unnecessary comparisons after each pass since the largest element will already be at the end of the array.